raphtory-graphql/src/model/graph/history.rs

```diff
@@ -0,0 +1,588 @@
+                .collect()
+        })
+        .await
+    }
+
+    /// Returns true if the History object is empty
+    async fn is_empty(&self) -> bool {
+        let self_clone = self.clone();
+        blocking_compute(move || self_clone.history.is_empty()).await
+    }
+
+    /// Returns the number of entries contained in the History object
+    async fn count(&self) -> u64 {
+        let self_clone = self.clone();
+        blocking_compute(move || self_clone.history.len() as u64).await
+    }
+
+    /// A HistoryTimestamp object which provides access to timestamps (as Unix epochs in milliseconds) instead of `TimeIndexEntry` entries
+    async fn timestamps(&self) -> GqlHistoryTimestamp {
+        let self_clone = self.clone();
+        blocking_compute(move || GqlHistoryTimestamp {
+            history_t: HistoryTimestamp::new(self_clone.history.0.clone()), // clone the Arc, not the underlying object
+        })
+        .await
+    }
+
+    /// A HistoryDateTime object which provides access to datetimes instead of `TimeIndexEntry` entries.
+    /// Uses RFC 3339 format by default (e.g., "2023-12-25T10:30:45.123Z").
+    /// Optionally takes a format string as argument to format the output of datetimes.
+    /// Refer to chrono::format::strftime for formatting specifiers and escape sequences.
+    async fn datetimes(&self, format_string: Option<String>) -> GqlHistoryDateTime {
+        let self_clone = self.clone();
+        blocking_compute(move || GqlHistoryDateTime {
+            history_dt: HistoryDateTime::new(self_clone.history.0.clone()), // clone the Arc, not the underlying object
+            format_string,
+        })
+        .await
+    }
+
+    /// A HistorySecondaryIndex object which provides access to the secondary indices of `TimeIndexEntry` entries
+    async fn secondary_index(&self) -> GqlHistorySecondaryIndex {
+        let self_clone = self.clone();
+        blocking_compute(move || GqlHistorySecondaryIndex {
+            history_s: HistorySecondaryIndex::new(self_clone.history.0.clone()), // clone the Arc, not the underlying object
+        })
+        .await
+    }
+
+    /// Returns an Intervals object that provides access to the intervals between temporal entries
+    async fn intervals(&self) -> GqlIntervals {
+        let self_clone = self.clone();
+        blocking_compute(move || GqlIntervals {
+            intervals: Intervals::new(self_clone.history.0.clone()), // clone the Arc, not the underlying object
+        })
+        .await
+    }
+}
+
+/// History object that provides access to timestamps (as Unix epochs in milliseconds) instead of `TimeIndexEntry` entries
+#[derive(ResolvedObject, Clone)]
+#[graphql(name = "HistoryTimestamp")]
+pub struct GqlHistoryTimestamp {
+    history_t: HistoryTimestamp<Arc<dyn InternalHistoryOps>>,
+}
+
+#[ResolvedObjectFields]
+impl GqlHistoryTimestamp {
+    /// A list of all timestamps (as Unix epochs in milliseconds) present in this history,
```

Comment:
```suggestion
    /// A list of all timestamps (as Unix epochs in milliseconds) present in this history.
```

In general missing sentence formating.